classes/service/post_service.php

```diff
@@ -151,6 +151,34 @@ public function handle_update_post($course, $cm, $forum, $context, $discussion,
             $this->db->set_field('hsuforum_discussions', 'groupid', $options['groupid'], array('id' => $discussion->id));
         }
 
+        // Handle templating if 'post to my groups' checkbox enabled.
+        if (!empty($posttomygroups)) {
+            $allowedgroups = groups_get_activity_allowed_groups($cm);
+            foreach ($allowedgroups as $groupid => $group) {
+                if (hsuforum_user_can_post_discussion($forum, $groupid, -1, $cm, $context)) {
+                    $groupstopostto[] = $groupid;
+                } else {
+                    $groupstopostto[] = $options['groupid'];
+                }
+            }
+
+            foreach ($groupstopostto as $groupid) {
```

Comment:
This looks like it might be looping over the same collection twice if `hsuforum_user_can_post_discussion()` is true for all groups. Can't this loop be removed and replaced by a method that gets called for every group id found in the loop above?

Reply:
This functionality that was added to post to all groups. It is tying into an already sloppy function not written by us. We should if effort is available refactor this.